hypothesis-python/src/hypothesis/internal/conjecture/datatree.py

```diff
@@ -178,14 +178,26 @@ def compute_max_children(ir_type, kwargs):
             return 1
         return 2
     elif ir_type == "bytes":
-        return 2 ** (8 * kwargs["size"])
+        min_size = kwargs["min_size"]
+        max_size = kwargs["max_size"]
+
+        if max_size is None:
+            max_size = COLLECTION_DEFAULT_MAX_SIZE
+
+        definitely_too_large = max_size * math.log(2**8) > math.log(
+            MAX_CHILDREN_EFFECTIVELY_INFINITE
+        )
+        if definitely_too_large:
+            return MAX_CHILDREN_EFFECTIVELY_INFINITE
+
+        return sum(2 ** (8 * k) for k in range(min_size, max_size + 1))
```

Comment:
I'd pull this and the similar strings logic out into a helper function, `n_distinct_strings(n_alphabet, min_size, max_size)`